dev/statement_serializer.h

```diff
@@ -52,6 +54,282 @@ namespace sqlite_orm {
             return serializer(t, context);
         }
 
+        inline void stream_identifier(std::ostream& ss,
+                                      const std::string& qualifier,
+                                      const std::string& identifier,
+                                      const std::string& alias) {
+            constexpr char quoteChar = '"';
+
+            if(!qualifier.empty()) {
+                ss << quoteChar << sql_escape(qualifier, quoteChar) << std::string{quoteChar, '.'};
+            }
+            ss << quoteChar << sql_escape(identifier, quoteChar) << quoteChar;
```

Comment:
Are you sure that identifiers can contain double quotes in SQLite?

Reply:
Yes, they can.

Reply:
ok. If we add check whether identifiers contain a double-quote what if it turns out that it does?

Reply:
Instead of `sql_escape()` we'll be using:
```c++
inline void stream_sql_escaped(std::ostream& os, const std::string& str, char char2Escape) {
    for (std::string::const_iterator it = str.cbegin(), next; true; it = next + 1) {
        next = std::find(it, str.cend(), char2Escape);
        os << std::string_view{ it, next };

        if (next == str.cend()) {
            break;
        }
        os << std::string(2, char2Escape);
    }
}
```

Reply:
BTW `std::string_view` is not available in C++14. Actually this code has a longer algorithm. I mean it will work longer at runtime. Let's leave it like that. I doubt that we will find a lot of people who uses double quotes in identifiers in their databases. So let's skip over-engineering. Probably we are afraid of a monster which doesn't exist. But of course if someone faces such an issue we'll upgrade the code with a snippet you provided. That do you think?

Reply:
Hmm, why do you worry about runtime overhead? I wouldn't sacrifice accuracy for a few nanoseconds of runtime performance. There are so many other places that chew up runtime.
One could even argue about security - think of sql injection. You are absolutely right about the rareness of such identifiers, but I am not "afraid" of a monster, rather think about accuracy.

So I am not giving up easily :) it's just about finding creative ways...

What about less over-engineered:
```c++
inline void stream_sql_escaped(std::ostream& os, const std::string& str, char char2Escape) {
    for (char c : str) {
        if (c == char2Escape) {
            os << char2Escape;
        }
        os << c;
    }
}
```
Or even better:
```c++
inline void stream_sql_escaped(std::ostream& os, const std::string& str, char char2Escape) {
    if (str.find(char2Escape) == str.npos) [[likely]] {
        os << str;
    } else {
        for (char c : str) {
            if (c == char2Escape) {
                os << char2Escape;
            }
            os << c;
        }
    }
}
```

Otherwise, if you still think of it as a "no", we should prohibit double-quotes in table and column identifiers at storage construction time.

Reply:
it looks like ok. I agree with your words. Let's add such a check